src/Gui/ReportView.cpp

```diff
@@ -170,6 +176,21 @@ void ReportHighlighter::highlightBlock (const QString & text)
 
         start = it->length;
     }
+    if(hasErrors || hasWarnings){
+        //activate report view unless this is disabled in preferences
+
+        ParameterGrp::handle group = App::GetApplication().GetUserParameter().
+        GetGroup("BaseApp")->GetGroup("Preferences")->GetGroup("OutputWindow");
+        bool showReportViewOnWarningOrError = group->GetBool("checkShowReportViewOnWarningOrError", true);
+        if (showReportViewOnWarningOrError){
+            QWidget* reportView = Gui::getMainWindow()->findChild<QWidget*>(QString::fromLatin1("Report view"));
+            if (reportView){

```

Comment:
Managing the visibility of the report window directly is cumbersome and causes inconsistencies in the GUI. If you look at the View menu and under Panels you see all the dock windows listed. A QDockWidget has a "toggle action" and it's best to manage the visibility over this.